File: 000 Daily/1.py
# LINK: https://leetcode.com/problems/remove-sub-folders-from-the-filesystem/

from typing import List


def removeSubfolders(folder: List[str]) -> List[str]:
    folder.sort()
    res = [folder[0]]
    
    for i in range(1, len(folder)):
        last_folder = res[-1] + '/'
        
        if not folder[i].startswith(last_folder):
            res.append(folder[i])
    
    return res

File: 000 Daily/1.rs
// LINK: https://leetcode.com/problems/remove-sub-folders-from-the-filesystem/

pub fn remove_subfolders(mut folder: Vec<String>) -> Vec<String> {
    folder.sort();
    let mut res = vec![folder[0].to_owned()];

    for i in 1..folder.len() {
        let last_folder = res.last().unwrap().to_owned() + "/";

        if !folder[i].starts_with(&last_folder) {
            res.push(folder[i].to_owned())
        }
    }
    
    res
}
